process/transaction/shardProcess.go

```diff
@@ -250,6 +248,28 @@ func (txProc *txProcessor) processTxFee(
 	return cost, nil
 }
 
+func (txProc *txProcessor) checkIfValidTxToMetaChain(
+	tx *transaction.Transaction,
+	acntSnd state.UserAccountHandler,
+	adrDst state.AddressContainer,
+) error {
+
+	destShardId := txProc.shardCoordinator.ComputeId(adrDst)
+	if destShardId != core.MetachainShardId {
+		return nil
+	}
+
+	// it is not allowed to send transactions to metachain if those are not of type smart contract
+	if len(tx.GetData()) == 0 {
+		receiptErr := txProc.executingFailedTransaction(tx, acntSnd, process.ErrInvalidMetaTransaction)
```

Comment:
Why not directly: return txProc.executingFailedTransaction(tx, acntSnd, process.ErrInvalidMetaTransaction) ?

Reply:
done.